plugins/hosts/windows/cap/os_arch.rb

```diff
@@ -0,0 +1,18 @@
+module VagrantPlugins
+  module HostWindows
+    module Cap
+      class OSArch
+        def self.os_arch(env)
+          # Logic taken from http://stackoverflow.com/a/25845488
```

Comment:
@hferentschik More reliable solution than this is welcome.

Below code works in Window's cmd, powershell, mingw shell and cygwin.